### ISSUE TITLE
[Flake] SEGFAULT in cloud-cpp/github/google-cloud-cpp/master/macos/cmake-super

### ISSUE THREAD

**Issue by devbww** [googleapis/google-cloud-cpp#4399]
http://sponge.corp.google.com/a2eafbff-44b3-4832-946d-e21c186af4b1

```
 34/263 Test  #31: common_grpc_utils_completion_queue_test ...........................***Exception: SegFault  0.21 sec
Running main() from gmock_main.cc
[==========] Running 14 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 14 tests from CompletionQueueTest
[ RUN      ] CompletionQueueTest.TimerSmokeTest
[       OK ] CompletionQueueTest.TimerSmokeTest (4 ms)
[ RUN      ] CompletionQueueTest.MockSmokeTest
[       OK ] CompletionQueueTest.MockSmokeTest (17 ms)
[ RUN      ] CompletionQueueTest.ShutdownWithPending
[       OK ] CompletionQueueTest.ShutdownWithPending (30 ms)
[ RUN      ] CompletionQueueTest.CanCancelAllEvents
[       OK ] CompletionQueueTest.CanCancelAllEvents (6 ms)
[ RUN      ] CompletionQueueTest.MakeUnaryRpc
[       OK ] CompletionQueueTest.MakeUnaryRpc (27 ms)
[ RUN      ] CompletionQueueTest.MakeStreamingReadRpc
[       OK ] CompletionQueueTest.MakeStreamingReadRpc (13 ms)
[ RUN      ] CompletionQueueTest.MakeRpcsAfterShutdown
[       OK ] CompletionQueueTest.MakeRpcsAfterShutdown (2 ms)
[ RUN      ] CompletionQueueTest.RunAsync
[       OK ] CompletionQueueTest.RunAsync (6 ms)
[ RUN      ] CompletionQueueTest.RunAsyncCompletionQueueDestroyed

:

99% tests passed, 1 tests failed out of 263

Total Test time (real) =  19.45 sec

The following tests FAILED:
	 31 - common_grpc_utils_completion_queue_test (SEGFAULT)
Errors while running CTest
make[2]: *** [build/google-cloud-cpp/src/google_cloud_cpp_project-stamp/google_cloud_cpp_project-test] Error 8
```

**Comment by coryan** [comment]
Labeling as "flake" as this does not happen on every run. Note that I have made several fixes to these tests, it is possible this flake is fixed, but might be hard to prove it is.

**Comment by coryan** [comment]
FWIW, I just ran that specific test 100,000 times on my macOS laptop, no crash.  The full test passed 300 times.

There are a number of fixes in this area since the bug was reported: #4427, #4448, #4450, and #4474. Of those,
the problem fixed by #4427 probably resulted in a segfault in any test using timers. While this test does not use timers directly, the function it calls uses a timer under the hood.

I am going to close this: I cannot repro, and there are fixes that look plausible.
